documentation/modules/ROOT/pages/ai/embeddings.adoc

```diff
@@ -0,0 +1,191 @@
+:page-aliases: ai/embeddings.adoc
+// Category: debezium-using
+// Type: assembly
+// ModuleID: embeddings-transformation
+// Title: Embeddings Transformation
+[id="embeddings-transformation"]
+= Embeddings Transformation
+ifdef::community[]
+:toc:
+:toc-placement: macro
+:linkattrs:
+:icons: font
+:source-highlighter: highlight.js
+
+toc::[]
+endif::community[]
+
+In large language models, or in natural language processing in general, one of the important tasks is conversion of the text to the numerical representation.
+One way how to do that is so-called text embeddings.
+It's a process of conversion of a text to a high-dimensional numerical vector.
+Representing text as a vector allows to do semantic similarity search and other advanced operations.
+
+{prodname} offers build-in feature which transforms specified field into embedding and add this text embedding into the record.
+Embedding inference is done by the selected model served by configured provider.
+{prodname} offers several embeddings model providers.
+
+Technically it's implemented as a  link:{link-kafka-docs}/#connect_transforms[single message transformation] (SMT) and under the hood it uses link:https://docs.langchain4j.dev/[langchain4j] framework to interact with embeddings providers.
+
+== Behavior
+
+Embeddings transformation takes specified record field and passes it to the configured embedding model for inference, i.e. creating an embedding from the text contained in this field.
+Resulting embedding is appended to the record.
+The source field is still part of the record.
+
+The source field has to be a string field.
+Embedding field is a vector of float 32-bits numbers.
+The size of the vector depends on the selected model.
+{prodname} provides link:https://github.com/debezium/debezium/blob/main/debezium-core/src/main/java/io/debezium/data/vector/FloatVector.java[FloatVector] data type, which is used for internal representation of the embedding.
+The schema type of the embedding field in the record is `io.debezium.data.FloatVector`.
+
+Both source and embedding field specification supports nested structures, e.g. `after.product_description_embedding`.
+
+== Configuration
+
+To create embeddings transformation, add SMT configuration to your connector configuration:
+
+[source]
+----
+transforms=ai,
+transforms.ai.type=io.debezium.ai.embeddings.FieldToEmbedding
+----
+
+You have to specify at least field which content will be used as an input for the embedding and the destination field where the embedding will be placed:
+
+[source]
+----
+ai.embeddings.field.source=after.product
```

Comment:
Shouldn't this be prefixed with `transforms.`?

Reply:
thanks @mfvitale , good point